runtime/src/bank.rs

```diff
@@ -111,6 +111,9 @@ pub struct Bank {
 
     /// The pubkey to send transactions fees to.
     collector_id: Pubkey,
+
+    /// if this is a tpu bank
+    pub tpu: bool,
```

Comment:
This seems way too context aware. Bank can be used by anything.  Take, for example, the field just above it. It's called a "collector", not a "leader". If we must have this boolean, can it instead be in the form `is_<capability>`?

Reply:
@garious what about sticking into forks?

Reply:
Sure, but that's the working bank, which I thought we were working to purge.

Reply:
@garious, I was thinking as metadata for the banks.

Reply:
Sounds like you need a `BankEntry` struct in `BankForks` that wraps `Bank` and adds metadata.